components/MonitoringUserSetupInstructions.tsx

```diff
@@ -261,5 +276,65 @@ $$ LANGUAGE sql VOLATILE SECURITY DEFINER;`}
   )
 }
 
+const NoPgMonitorPgStatStatementsHelpers: React.FunctionComponent<{adminUsername?: string}> = ({adminUsername}) => {
+  const adminUserStr = !!adminUsername ? <strong>{adminUsername}</strong> : 'a superuser (or equivalent)'
+  const CodeBlock = useCodeBlock();
+  return (
+    <>
+      <p>
+        Because your Postgres provider does not give you access to the <code>pg_monitor</code>
+        role that can read all queries from pg_stat_statements, you'll need to create a set of
+        SECURITY DEFINER helper functions to collect full query stats.
+      </p>
+      <p>
+        For each user whose queries you would like to monitor, you'll need to log in to your
```

Comment:
Maybe. I tried to keep this parallel with the self-managed instructions for easier maintenance. I'll review what we can do here.

Reply:
Move that to its own step, and it flows better in addition to avoiding errors. Thanks for catching this. I originally moved this up after going through setup and thought it was safe because the function body is not evaluated until later, but the return type depends on the extension, too. Having this as its own step works better.